Automattic/jetpack
# Handle tree dropdown visible area click

Fixes https://github.com/Automattic/dotcom-forge/issues/5798

On the `TreeDropdown` component, only the text input is clickable.

![image](https://github.com/Automattic/jetpack/assets/33497086/646d23ed-2cd2-4163-a230-b54561a8953b)

But the intent behavior is that the yellow area below also expands the categories tree and focuses on the input:
![image](https://github.com/Automattic/jetpack/assets/33497086/b2bff02a-f275-45f6-82c5-d399ad9e3216)

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
To enable the feature flag navigate with **enable-newsletter-categories** query param:
`wp-admin/admin.php?enable-newsletter-categories=true&page=jetpack#/newsletter`

- Enable the newsletter categories toggle.
- Click on any part of the `TreeDropdown` represented as yellow in the image above. It should expand the categories tree and focus the input.
- Clicking on the close X icon of the categories pill should remove the category, but not trigger the focus of the input element..
